opensearch-operator/api/v1/opensearch_index_types.go

```diff
@@ -4,6 +4,16 @@ import (
 	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 )
 
+type OpensearchDataStreamTimestampFieldSpec struct {
+	// Name of the field that are used for the DataStream
+	Name string `json:"name"`
+}
+
+type OpensearchDataStreamSpec struct {
+	// TimestampField for dataStream
+	TimestampField OpensearchDataStreamTimestampFieldSpec `json:"timestamp_field,omitempty"`
```

Comment:
Sorry for not being clearer, my camelCase comment was primarily aimed at `timestamp_field` (should be `timestampField`). Even though in the opensearch api it is snake_case, the operator CRDs use camelCase all around